airbyte-integrations/connectors/source-slack/source_slack/source.py

```diff
@@ -209,8 +209,9 @@ def read_records(
         stream_state: Mapping[str, Any] = None,
     ) -> Iterable[Mapping[str, Any]]:
         if not stream_slice:
+            # return an empty iterable
             # this is done to emit at least one state message when no slices are generated
-            return []
+            return iter([])
```

Comment:
```suggestion
            yield from []
```

Reply:
Both work, but suggestion based on how we handle this case in the CDK elsewhere (`http.py`, `simple_retriever.py`)